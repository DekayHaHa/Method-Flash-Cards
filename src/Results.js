import React from 'react';

export default function Results (props) {
		const results = props.answer ? 'CORRECT!' : 'JUST DO BETTER NEXT TIME!';
		return (
			<section>
				<h3>{results}</h3>
				<p>Here is a link to the <a target='_blank' rel="noopener noreferrer" href={props.prompt.mdn}>MDN</a> on</p> <p className='method-name'>'{props.prompt.solution}'</p>
				<button className='continue' onClick={props.close}>Continue</button>
			</section>
		);
	}
