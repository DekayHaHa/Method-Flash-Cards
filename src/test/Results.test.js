import React from 'react';
import ReactDOM from 'react-dom';
import Results from '../Results';
import { shallow } from 'enzyme';

const closeMock = jest.fn();
const mockPrompt = {
	"snippet": ["let arr = [1, 2, 3, 4, 5]", "let results = arr.", "(val => val > 0)", "console.log(results) // logs true"],
	"category": "Array Prototype Method",
	"solution": "every",
	"mdn": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every",
	"uid": "card-1"
}

describe('Results', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = shallow(
			<Results answer={false} close={closeMock} prompt={mockPrompt}/>
		);
	});
	it('should match snapshop', () => {
		expect(wrapper).toMatchSnapshot();
	});
	
	it('should invoke props.close on click of button', () => {
		wrapper.find('.continue').simulate('click');
		expect(closeMock).toHaveBeenCalled();
	})
})