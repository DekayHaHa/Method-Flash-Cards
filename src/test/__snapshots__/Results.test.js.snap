// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should match snapshop 1`] = `
<section>
  <h3>
    JUST DO BETTER NEXT TIME!
  </h3>
  <p>
    Here is a link to the 
    <a
      href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every"
      rel="noopener noreferrer"
      target="_blank"
    >
      MDN
    </a>
     on
  </p>
   
  <p
    className="method-name"
  >
    '
    every
    '
  </p>
  <button
    className="continue"
    onClick={[MockFunction]}
  >
    Continue
  </button>
</section>
`;
